vmm/src/lib.rs

```diff
@@ -778,6 +782,41 @@ impl Vmm {
         self.kernel_config = Some(kernel_config);
     }
 
+    fn flush_metrics_on_request(&mut self) -> std::result::Result<VmmData, VmmActionError> {
+        if let Err(e) = self.write_metrics() {
+            if let LoggerError::NeverInitialized(s) = e {
+                return Err(VmmActionError::Logger(
+                    ErrorKind::User,
+                    LoggerConfigError::FlushMetrics(s),
+                ));
+            } else {
+                return Err(VmmActionError::Logger(
+                    ErrorKind::Internal,
+                    LoggerConfigError::FlushMetrics(e.to_string()),
+                ));
+            }
+        }
+        Ok(VmmData::Empty)
+    }
+
+    fn flush_metrics_on_event(&mut self) {
+        self.write_metrics_event.fd.read();
+        // Please note that, since LOGGER has no output file configured yet, it will write to
+        // stdout, so logging will interfere with console output.
+        if let Err(e) = self.write_metrics() {
+            warn!("Failed to log metrics: {}", e);
```

Comment:
In the case where the logger is not configured/initialized, this error will spam the console every 60s, and it's not even an error.

I think the whole logic around `LoggerError::NeverInitialized` needs to change since we decided to allow logging output to go to stdout prior to logger user configuration.

I think the way to go is to simply not consider the `NeverInitialized` case an error. The only place that should care if logger was initialized or not would be the `ActionType::FlushMetrics` code-path (it's an user error to request metrics without configuring the metrics pipe).

Reply:
+1. Separating the errors as I suggested in my comment from the previous commit should help with this as well.